### PR TITLE
fix (FOROME-449): the subattributes list sometimes isnt full

### PR DESCRIPTION
**Pull request by YaroslavTrefilovBraveDevelopers**

### CODE REVIEW

File: src/pages/filter/ui/selected-group-items.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { ReactElement } from 'react'
+import Checkbox from 'react-three-state-checkbox'
+import { observer } from 'mobx-react-lite'
+
+import { IVariantList } from '@declarations'
+import filterStore from '@store/filter'
+
+interface IProps {
+  chunks: any[]
+  currentPage: number
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/selected-group-items.tsx:10]: You only need 'current page' for the index to access the item. Why not pass the actual item of the 'chunks'?

File: src/pages/filter/ui/selected-group-items.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      <div className="mt-4">
+        {chunks[currentPage]?.map((element: IVariantList, index: number) => {
+          const { variant, isChecked } = element
+          const name = variant[0]
+          const amount = variant[1]
+
+          return (
+            <div
+              key={name}
+              className="flex items-center mb-3 text-14  leading-16px"
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/selected-group-items.tsx:26]: double spacing

File: src/pages/filter/ui/selected-group-items.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+              <Checkbox
+                checked={isChecked}
+                style={{ cursor: 'pointer' }}
+                indeterminate={filterStore.selectedGroupItem.name === name}
+                onChange={event => {
+                  handleSelect(index, event.target.checked)
+                }}
+              />
+
+              <span key={name} className="font-medium ml-2">
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/selected-group-items.tsx:37]: Why do you need a key here?

File: src/store/dataset.ts [hunk 4]
```diff
@@ -158,13 +158,16 @@ class DatasetStore {
...
-    subGroup: string
-    itemName: string
-  }) {
+  removeCondition(
+    {
+      subGroup,
+      itemName,
+    }: {
+      subGroup: string
+      itemName: string
```
Comment (AndreyUstyumenko) [inline src/store/dataset.ts:167]: better to move to index.d.ts

File: src/pages/filter/ui/enum-panel.tsx [hunk 5]
```diff
@@ -1,76 +1,160 @@
...
+    }, [stringedVariants])
+
+    const handleClear = () => {
+      const localVariantList = variantList.map(element => {
+        element.isChecked = false
+
+        return element
+      })
+
+      setVariantList(localVariantList)
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/enum-panel.tsx:71]: I suppose, the selected filters should be cleared on the very left panel as well. At least, other filters behave like that.

File: src/pages/filter/ui/selected-group-items.tsx (hunk 1 above)
Comment (YaroslavTrefilovBraveDevelopers) [reply src/pages/filter/ui/selected-group-items.tsx]: Great point, I have fixed

File: src/pages/filter/ui/selected-group-items.tsx (hunk 2 above)
Comment (YaroslavTrefilovBraveDevelopers) [reply src/pages/filter/ui/selected-group-items.tsx]: I fixed

File: src/pages/filter/ui/selected-group-items.tsx (hunk 3 above)
Comment (YaroslavTrefilovBraveDevelopers) [reply src/pages/filter/ui/selected-group-items.tsx]: Apparently it was left from the previous implementation, but I forgot to remove.  I fixed

File: src/store/dataset.ts (hunk 4 above)
Comment (YaroslavTrefilovBraveDevelopers) [reply src/store/dataset.ts]: Moved

File: src/pages/filter/ui/enum-panel.tsx (hunk 5 above)
Comment (YaroslavTrefilovBraveDevelopers) [reply src/pages/filter/ui/enum-panel.tsx]: I agree. Fixed

**Review by Lapk1n — APPROVED** [review]

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**